### PR TITLE
Support `Theme` attribute in `Code Block` macro

### LINKED ISSUE #810: Support `Theme` attribute in `Code Block` macro (v1 editor)
Hi.  Thank you for the very useful package!

As the title suggests, I'm looking to have support for the `Theme` attribute in `Code Block` macros.  I see this as an optional setting in `conf.py` (e.g. `confluence_code_block_theme`) that would apply to all `Code Block` macros in the rst file(s).

I'm happy to have a go at implementing this myself, if it's something you would consider merging.  Thanks!

### PR DESCRIPTION
**Pull request by bpandola**
Closes #810

### CODE REVIEW

**Review by jdknight — CHANGES_REQUESTED** [review]
Brian, thanks for taking the time to making a pull request for this. Going to dump a series of review comments on if (mostly nitpicky stuff), if you do not mind address them ~ if not, I can help look at this a later time.

Aside from the review comments, the one thing I see missing is a configuration check entry for the new option. At minimum, we should have a check added to verify the string type of this configuration's value. For example, consider how `confluence_space_key` is [checked](https://github.com/sphinx-contrib/confluencebuilder/blob/v2.1.1/sphinxcontrib/confluencebuilder/config/checks.py#L706-L712).

Lastly, and completely optional, if you want to be extremely picky for the theme configuration option, you may even want to define a list of all acceptable theme types supported by Confluence (listing in somewhere in our ["Confluence standards" area](https://github.com/sphinx-contrib/confluencebuilder/blob/main/sphinxcontrib/confluencebuilder/std/confluence.py), and generate warnings if a user provides a theme that is not supported by Confluence. For sure not required; and if missing, will not block this pull request.

File: sphinxcontrib/confluencebuilder/storage/translator.py [hunk 1]
```diff
@@ -755,10 +755,17 @@ def visit_literal_block(self, node):
...
+        theme = None
+        if self.builder.config.confluence_code_block_theme:
+            theme = self.builder.config.confluence_code_block_theme
+
         self.body.append(self._start_ac_macro(node, 'code'))
         self.body.append(self._build_ac_param(node, 'language', lang))
         self.body.append(self._build_ac_param(node, 'linenumbers', num))
 
+        if theme is not None:
+            self.body.append(self._build_ac_param(node, 'theme', theme))
```
[the remark this answers is not in the document]
Comment (bpandola) [reply sphinxcontrib/confluencebuilder/storage/translator.py:767]: I think I was trying to follow what appeared to be an existing convention (e.g. with `title` and `firstline`), but I'm happy to simplify per your suggestion (as I don't see a need to leave `theme` exposed).

File: tests/unit-tests/test_sphinx_codeblock.py [hunk 2]
```diff
@@ -28,6 +29,22 @@ def test_storage_sphinx_codeblock_caption(self):
             self.assertIsNotNone(title_param)
             self.assertEqual(title_param.text, 'code caption test')
 
+    @setup_builder('confluence')
+    def test_storage_sphinx_codeblock_theme(self):
+        theme = 'Midnight'
+        config = prepare_conf()
+        config.confluence_code_block_theme = theme
```
[the remark this answers is not in the document]
Comment (bpandola) [reply tests/unit-tests/test_sphinx_codeblock.py:36]: Thanks.  I looked, but clearly not hard enough, for a similar example.  I appreciate you pointing it out.

**Comment by bpandola** [comment]
@jdknight Thanks for the review.  I think I've addressed everything you mentioned.  I'll keep adding commits until you approve, but let me know if I need to squash into one commit (with the DCO sign-off-by) before you merge.

**Comment by jdknight** [comment]
@bpandola, thanks for addressing each point (and more). It looks great. Once the DCO is addressed, we should be able to merge this in.

**Comment by bpandola** [comment]
@jdknight DCO addressed.

**Review by jdknight — APPROVED** [review]

**Pull request merged**